### PR TITLE
bucket polish and enable evict-cache-on-memory-ratio by default

### LINKED ISSUE #14356: [Dynamic Regions] tikv oom due to memtable is highly used
### What version of TiKV are you using?
./tikv-server -V
 TiKV 
Release Version:   6.7.0-alpha
Edition:           Community
Git Commit Hash:   8bfa12dea2d5bd0918b35e5ae041676aeeee77ab
Git Commit Branch: heads/refs/tags/v6.7.0-alpha
UTC Build Time:    2023-03-05 11:40:05
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure
Profile:           dist_release
2023-03-06T07:00:46.903Z	INFO	k8s/client.go:132	it should be noted that a long-running command will not be interrupted even the use case has ended. For more information, please refer to https://github.com/pingcap/test-infra/discussions/129

### What operating system and CPU are you using?
8c/32G

### Steps to reproduce
https://tcms.pingcap.net/dashboard/executions/plan/1654853
after case ha_tikv(random)_io_delay(10ms), tikv oom multi times
http://perf.pingcap.net:3000/d/C1q8KDMVz/ha-test-summary?orgId=1&var-testbed=endless-ha-test-multi-rocks-tps-1654853-1-283&var-tag=All&var-case_name=All

### What did you expect?
tikv should not be oom

### What did happened?
3 tikvs oom
logs: http://minio.pingcap.net:38888/buckets/test-infra-testground/browse/YXJjaGl2ZS9lbmRsZXNzLWhhLXRlc3QtbXVsdGktcm9ja3MtdHBzLTE2NTQ4NTMtMS0yODMv
clinic:  https://clinic.pingcap.com.cn/portal/#/login?next=%2Forgs%2F31%2Fclusters%2F7207301522984517045
image
image
774f0822-2aec-4b13-9cbc-c129b94fe008

### PR DESCRIPTION
**Pull request by tonyxuqqi**
### What is changed and how it works?

Issue Number: Close #14356

What's Changed:

```commit-message
1) avoid sending PeerTick::SplitRegionCheck for each write
2) reduce default bucket size from 96MB to 50MB to reduce chance of client side timeout
3) change the default value of evict-cache-on-memory-ratio to 0.1. 
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- No code

### Release note 

```release-note
1) reduce default bucket size from 96MB to 50MB to reduce chance of client side timeout
2) change the default value of evict-cache-on-memory-ratio to 0.1. 
```

### CODE REVIEW

File: components/raftstore/src/coprocessor/config.rs [hunk 1]
```diff
@@ -76,7 +76,7 @@ pub const RAFTSTORE_V2_SPLIT_SIZE: ReadableSize = ReadableSize::gb(10);
 /// Default batch split limit.
 pub const BATCH_SPLIT_LIMIT: u64 = 10;
 
-pub const DEFAULT_BUCKET_SIZE: ReadableSize = ReadableSize::mb(96);
+pub const DEFAULT_BUCKET_SIZE: ReadableSize = ReadableSize::mb(50);
```
Comment (overvenus) [inline components/raftstore/src/coprocessor/config.rs:79]: Could you add some comments about 50MiB?

File: components/raftstore-v2/src/operation/bucket.rs [hunk 2]
```diff
@@ -246,6 +270,15 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
...
 
         let buckets_count = region_buckets.meta.keys.len() - 1;
+        if change_bucket_version {
+            // TODO: we may need to make it debug once the coprocessor timeout is resolved.
+            info!(self.logger, "refreshed region bucket info";
+                "bucket_version" => next_bucket_version,
+                "buckets_count" => buckets_count,
+                "estimated_region_size" => region_buckets.meta.total_size(),
+                "first_refresh" => is_first_refresh,
+            );
```
Comment (SpadeA-Tang) [inline components/raftstore-v2/src/operation/bucket.rs:280]:
```suggestion
            info!(
                self.logger, 
                "refreshed region bucket info";
                "bucket_version" => next_bucket_version,
                "buckets_count" => buckets_count,
                "estimated_region_size" => region_buckets.meta.total_size(),
                "first_refresh" => is_first_refresh,
            );
```

File: components/raftstore/src/coprocessor/config.rs [hunk 3]
```diff
@@ -76,7 +76,11 @@ pub const RAFTSTORE_V2_SPLIT_SIZE: ReadableSize = ReadableSize::gb(10);
 /// Default batch split limit.
 pub const BATCH_SPLIT_LIMIT: u64 = 10;
 
-pub const DEFAULT_BUCKET_SIZE: ReadableSize = ReadableSize::mb(96);
+// A bucket will be split only when its size is larger than 2x of
+// DEFAULT_BUCKET_SIZE So the avg of the actual bucket size is 75MB, which is
+// slightly less than region size We don't use 48MB size because it will enable
+// the automatic bucket under default 96MB region size.
```
Comment (SpadeA-Tang) [inline components/raftstore/src/coprocessor/config.rs:82]: what you mean automatic bucket? I remember the bucket is not enabled if dynamic region is not set.

**Review by SpadeA-Tang — APPROVED** [review]

File: components/raftstore/src/coprocessor/config.rs (hunk 3 above)
Comment (tonyxuqqi) [reply components/raftstore/src/coprocessor/config.rs:82]: No, bucket is enabled automatically when bucket size * 2 <= region size, regardless dynamic region or not @SpadeA-Tang 

  if let Ok(()) = res && self.enable_region_bucket.is_none() {
            let useful = self.region_split_size() >= self.region_bucket_size * 2;
            self.enable_region_bucket = Some(useful);
        } else if let Err(e) = res && self.enable_region_bucket() {
            return Err(e);
        }

**Review by overvenus — APPROVED** [review]

**Pull request merged**